### PR TITLE
Check userOpReceipt for reverts

### PR DESCRIPTION
**Pull request by dospore**
Alan forwarded a request to return `success: false` if the internal userOp reverts but the transaction succeeds.

Note: Alchemy doesnt have a waitForReceipt, but their function actually just waits for the receipt and then only returns the txHash, kind of annoying but I just created a new func to try and avoid making side effects.

Viem waitForUserOperationReceipt is much slower as alchemy queries the rundler directly. This also doesnt add any latency since its just returning extra information from waitForUserOperationTransaction

### CODE REVIEW

File: patches/@aa-sdk+core+4.0.0-beta.9.patch
Comment (arjun-io) [inline patches/@aa-sdk+core+4.0.0-beta.9.patch:1]: Should we make this patch part of a separate PR?

File: src/relayer/index.ts [hunk 1]
```diff
@@ -80,6 +80,7 @@ export async function createRelayer() {
       }
     }
 
+    let erroredTxHash
```
Comment (arjun-io) [inline src/relayer/index.ts:83]: Can we just use a single `txHash` var here?

File: src/relayer/index.ts [hunk 2]
```diff
@@ -164,7 +166,12 @@ export async function createRelayer() {
               }
             })
               .catch(injectUOError(UOError.FailedWaitForOperation))
+            txHash = hash
             console.log(`UserOp confirmed: ${txHash}`)
+            if (userOpReceipt?.success === false) {
```
Comment (arjun-io) [inline src/relayer/index.ts:171]:
```suggestion
            if (!userOpReceipt?.success) {
```

File: patches/@aa-sdk+core+4.0.0-beta.9.patch
Comment (dospore) [reply patches/@aa-sdk+core+4.0.0-beta.9.patch:1]: This one depends on on the added `waitForUserOperationReceipt`

File: src/relayer/index.ts (hunk 1 above)
Comment (dospore) [reply src/relayer/index.ts]: Ive lifted the txHash var outside the try/catch so we can access it in the catch block

**Review by arjun-io — APPROVED** [review]
